File: Modul-6/Soal 4/PRAK604-2410817110004-MuhammadNaufalKhalish.py
kode=input()
pesan=input()
print(" ")
if len(kode)!=len(pesan):
    print("Panjang Kalimat berbeda, pesan palsu")
else :
    bintang=0
    pagar=0
    hasil=""

    for i,j in zip(kode,pesan):
        if i==" " and j==" ":
            continue

        if i==j:
            hasil +="*"
            bintang +=1

        else :
            hasil+="#"
            pagar +=1

    print(hasil)
    print(f"*={bintang}")
    print(f"#={pagar}")

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

File: Modul-6/Soal 4/PRAK604-2410817110004-MuhammadNaufalKhalish.c
#include <stdio.h>
#include <string.h>

int main(){
char kode[100],pesan[100];
int bintang=0,pagar=0;

fgets(kode, sizeof(kode), stdin);
fgets(pesan, sizeof(pesan), stdin);

kode[strcspn(kode, "\n")]= '\0';
pesan[strcspn(pesan, "\n")]= '\0';

if(strlen(kode) != strlen(pesan)){
    printf("Panjang Kalimat berbeda, pesan palsu");
    return 0; }

int panjang=strlen(kode);
for(int i=0; i<panjang; i++){
    if(kode[i]==' ' && pesan[i]==' '){
        continue;
    }

    if(kode[i]==pesan[i]){
        printf("*");
        bintang++;}
        
    else{ printf("#");
    pagar++;}
}

printf("\n *=%d",bintang);
printf("\n #=%d \n",pagar);

if(bintang>=pagar){
    printf("Pesan Asli \n");}
else{printf("Pesan Palsu \n");}

return 0;
}